src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -191,19 +193,23 @@ public void serverPacketData(INetworkInfo manager, AppEngPacket packet, EntityPl
 									}
 
 									// If that doesn't work, grab from the player's inventory
-									if ( whichItem == null && playerInventory != null ) {
+									if ( whichItem == null && playerInventory != null )
+									{
+										ItemStack playerItemStack = null;
 										for ( int y = 0; y < playerInventory.getSizeInventory(); y++ )
 										{
 											// Put the item in the test inventory, and check if the recipe is satisfied
-											testInv.setInventorySlotContents(x, playerInventory.getStackInSlot(y));
-											if ( r.matches(testInv, pmp.worldObj) )
+											playerItemStack = playerInventory.getStackInSlot( y );
```

Comment:
Changed. That diff is also a great example for the method is nested way too deep.